### PR TITLE
MKL usage example

### PR DESCRIPTION
**Pull request by Carl-Van**
Example using MKL.
Beware, MKL does not offer any wall time advantage in terms of Wall time when running on typical PC hardware.

### CODE REVIEW

**Comment by ddemidov** [comment]
Looks like the `mkl.hpp` is still missing in this PR.

**Comment by Carl-Van** [comment]
Denis,  Yes I did a PR on the example alone, and then I struggle to update the PR and/or create a new PR with mkl.hpp.
I tried hard but I don't seem to be able to find the light at the end ! :)

File: mkl.hpp [hunk 1]
```diff
@@ -0,0 +1,292 @@
+#ifndef AMGCL_BACKEND_MKL_HPP
+#define AMGCL_BACKEND_MKL_HPP
+
+/*
+The MIT License
+
+Copyright (c) 2012-2014 Denis Demidov <dennis.demidov@gmail.com>
```
Comment (ddemidov) [inline mkl.hpp:7]: Would you add your name here?

**Comment by ddemidov** [comment]
This looks fine now (I can see both files). I can add the example to the `CMakeLists.txt` later.

**Comment by ddemidov** [comment]
Thanks, merged!

**Comment by ilyapopov** [comment]
Should mkl.hpp go into amgcl directory instead of root directory?

**Comment by ddemidov** [comment]
It definitely should. Fixed in 7bd9448ae9a7f59cfbe03e7e9c334156fac04450, thanks!

**Comment by ddemidov** [comment]
See discussion on this PR: https://groups.google.com/g/amgcl/c/20qWFC9XOgs

**Pull request merged**